routemaster/feeds.py

```diff
@@ -0,0 +1,51 @@
+"""Creation and fetching of feed data."""
+from typing import Dict
+
+import requests
+
+from routemaster.utils import get_path
+
+
+def feeds_for_state_machine(state_machine) -> Dict[str, 'Feed']:
+    """Get a mapping of feed prefixes to unfetched feeds."""
+    return {
+        x.name: Feed(x.url, state_machine.name)
+        for x in state_machine.feeds
+    }
+
+
+class FeedNotFetched(Exception):
+    """Raised if we try to access the data in a feed we haven't yot fetched."""
+    pass
+
+
+class Feed:
+    """A feed fetcher, able to retreive a feed and read keys out of it."""
+
+    def __init__(self, url, state_machine):
+        """Create an un-fetched data feed."""
+        self.url = url
+        self.data = None
+        self.state_machine = state_machine
+
+    def fetch(self, label: str):
+        """Trigger the fetching of a feed's data."""
+        if self.data is not None:
+            return
+
+        url = self.url.replace(
+            '<label>',
+            label,
+        ).replace(
+            '<state_machine>',
+            self.state_machine,
+        )
+
+        response = requests.get(url)
```

Comment:
What behaviour do we want if this errors? (Should this `raise_for_status`?)

Reply:
Yep, this should fail for now. We might want to allow for more in the future, maybe 404 should be allowed for example. 473539d